x-pack/plugin/rank-rrf/src/internalClusterTest/java/org/elasticsearch/xpack/rank/rrf/RRFRankCoordinatorCanMatchIT.java

```diff
@@ -0,0 +1,257 @@
+            .setQueries(
+                List.of(
+                    new SearchQueryWrapperBuilder(QueryBuilders.rangeQuery("@timestamp").gt(495).lte(499)),
+                    new SearchQueryWrapperBuilder(QueryBuilders.rangeQuery("@timestamp").gte(500).lt(505))
+                )
+            )
+            .setSize(5)
+            .get();
+
+        assertNull(response.getHits().getTotalHits());
+        assertEquals(5, response.getHits().getHits().length);
+        assertEquals(5, response.getSuccessfulShards());
+        assertEquals(3, response.getSkippedShards());
+
+        // match 2 shards with overlap in queries
+        response = client().prepareSearch("time_index")
+            .setSearchType(SearchType.QUERY_THEN_FETCH)
+            .setPreFilterShardSize(1)
+            .setRankBuilder(new RRFRankBuilder(20, 1))
+            .setTrackTotalHits(false)
+            .setQueries(
+                List.of(
+                    new SearchQueryWrapperBuilder(QueryBuilders.rangeQuery("@timestamp").gt(495).lte(505)),
+                    new SearchQueryWrapperBuilder(QueryBuilders.rangeQuery("@timestamp").gte(497).lt(507))
+                )
+            )
+            .setSize(5)
+            .get();
+
+        assertNull(response.getHits().getTotalHits());
+        assertEquals(5, response.getHits().getHits().length);
+        assertEquals(5, response.getSuccessfulShards());
+        assertEquals(3, response.getSkippedShards());
+
+        // match one shard with one query in range and one query out of range
+        response = client().prepareSearch("time_index")
+            .setSearchType(SearchType.QUERY_THEN_FETCH)
+            .setPreFilterShardSize(1)
+            .setRankBuilder(new RRFRankBuilder(20, 1))
+            .setTrackTotalHits(false)
+            .setQueries(
+                List.of(
+                    new SearchQueryWrapperBuilder(QueryBuilders.rangeQuery("@timestamp").gt(501).lte(505)),
+                    new SearchQueryWrapperBuilder(QueryBuilders.rangeQuery("@timestamp").gte(10000).lt(10005))
+                )
+            )
+            .setSize(5)
+            .get();
+
+        assertNull(response.getHits().getTotalHits());
+        assertEquals(4, response.getHits().getHits().length);
+        assertEquals(5, response.getSuccessfulShards());
+        assertEquals(4, response.getSkippedShards());
+
+        // match no shards, but still use one to generate a search response
+        response = client().prepareSearch("time_index")
+            .setSearchType(SearchType.QUERY_THEN_FETCH)
+            .setPreFilterShardSize(1)
+            .setRankBuilder(new RRFRankBuilder(20, 1))
+            .setTrackTotalHits(false)
+            .setQueries(
+                List.of(
+                    new SearchQueryWrapperBuilder(QueryBuilders.rangeQuery("@timestamp").gt(4000).lte(5000)),
+                    new SearchQueryWrapperBuilder(QueryBuilders.rangeQuery("@timestamp").gte(10000).lt(10005))
```

Comment:
this is a good test! thanks for coming up with it. There is no chance that this can be a more contained test around can match? We have similar tests in `SearchServiceTests` I think

Reply:
This mimics portions of the can match tests for CCS. It's tricky to override the time based portion of collecting the specific `@timestamp` field, and this allows for that behavior fairly explicitly.